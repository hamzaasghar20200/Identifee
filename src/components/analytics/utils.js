import { floorFigure, parsePercentage, setDateFormat } from '../../utils/Utils';

export const formatQueryResult = (field, result, annotation, opts = {}) => {
  const { measures, dimensions } = annotation;

  if (!measures[field] && !dimensions[field]) {
    return result[field];
  }

  const attrs = measures[field] || dimensions[field];

  if (!attrs) {
    return result[field];
  }

  if (attrs.format === 'currency') {
    return floorFigure(result[field]);
  }
  if (attrs.format === 'percent') {
    return parsePercentage(result[field]);
  }
  if (attrs.type === 'time') {
    return setDateFormat(result[field], opts.format);
  }
  if (
    attrs.type === 'number' &&
    attrs.meta &&
    attrs.meta.type === 'date-diff'
  ) {
    return result[field].days || 0;
  }
  return result[field];
};
