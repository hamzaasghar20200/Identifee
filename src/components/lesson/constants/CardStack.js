export const stacks = [
  {
    id: 1,
    top: -5,
    left: 0,
  },
  {
    id: 2,
    top: -10,
    left: 0,
  },
  {
    id: 3,
    top: -15,
    left: 0,
  },
];
