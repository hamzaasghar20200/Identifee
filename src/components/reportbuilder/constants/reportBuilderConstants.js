import React from 'react';
import TextBlock from '../blocks/TextBlock';
import BlockDescription from '../blocks/BlockDescription';
import PartnerLogoBlock from '../blocks/PartnerLogoBlock';
import IconTextBlock from '../blocks/IconTextBlock';
import CircleChartTextBlock from '../blocks/CircleChartTextBlock';
import CalendarBlock from '../blocks/CalendarBlock';
import RPMGBlock from '../blocks/RPMGBlock';
import RawHTMLBlock from '../blocks/RawHTMLBlock';

export const ComponentTextTypes = {
  Donut: 'donut',
  Calendar: 'calendar',
  PercentText: 'percentText',
  IconText: 'iconText',
  DonutSelection: 'donutSelection',
  Static: 'static',
};

export const ComponentSourceTypes = {
  RPMG: 'rpmg',
  SPGlobal: 'spGlobal',
  FasterPayments: 'fasterPayments',
  Custom: 'custom',
};

export const PartnerLogoBySource = {
  rpmg: '/img/integrations/RPMG-Logo.png',
  spGlobal: '/img/integrations/sp-global.png',
  fasterPayments: '/img/integrations/FasterPayments-Logo.jpeg',
  custom: '',
};

export const ReportBlockControls = [
  {
    id: 3,
    icon: 'add_circle',
    type: 'addNew',
    tooltip: 'Insert a new block',
    isLoading: false,
  },
  {
    id: 5,
    icon: 'swap_horiz',
    type: 'changePosition',
    tooltip: 'Revert',
    isLoading: false,
  },
  {
    id: 6,
    icon: 'block',
    type: 'remove',
    tooltip: 'Disable',
    isLoading: false,
  },
];

export const ReportBlocksType = [
  { id: 1, key: 'All', name: 'All Blocks' },
  { id: 2, key: ComponentSourceTypes.SPGlobal, name: 'S&P Global' },
  { id: 3, key: ComponentSourceTypes.RPMG, name: 'RPMG' },
  { id: 4, key: ComponentSourceTypes.FasterPayments, name: 'Faster Payments' },
  { id: 5, key: ComponentSourceTypes.Custom, name: 'Custom' },
];

export const defaultReportBlocks = [
  {
    id: 1,
    direction: 'flex-row-reverse',
    description: [
      'The average Days Sales Outstanding (DSO) of your peer group.',
    ],
    data: {
      value: '64',
      text: 'Days',
    },
    type: 'Calendar',
    partnerLogo: {
      src: '/img/integrations/sp-global.png',
      position: 'float-right',
      style: { width: 100 },
    },
  },
  {
    id: 2,
    direction: '',
    description: [
      'On average, your peers pay (36%) of all payables between $2,500 - $10,000 by commercial card, providing a substantial increase in Days Payables.',
      'Peers are leveraging commercial card to improve Working Capital.',
    ],
    data: {
      labels: ['Card', 'Others'],
      data: {
        data: [36, 64],
        backgroundColor: ['#5cb85c', '#eee'],
      },
    },
    type: 'RPMG',
    partnerLogo: {
      src: '/img/integrations/RPMG-Logo.png',
      position: 'float-left',
    },
    options: { cutout: 40 },
  },
  {
    id: 3,
    direction: 'flex-row-reverse',
    description: [
      'A savings of $2.50 per item can be realized by moving from checks to digital payments.',
    ],
    data: {
      value: '$2.50',
      text: 'Savings',
    },
    type: 'Text',
    partnerLogo: {
      src: '/img/integrations/FasterPayments-Logo.jpeg',
      position: 'float-right',
    },
  },
  {
    id: 4,
    direction: '',
    description: [
      'The average Days Sales Outstanding (DSO) of your peer group.',
    ],
    data: {
      value: '45',
      text: 'Days',
    },
    type: 'Calendar',
    partnerLogo: {
      src: '/img/integrations/sp-global.png',
      position: 'float-left',
      style: { width: 100 },
    },
  },
  {
    id: 5,
    direction: 'flex-row-reverse',
    description: [
      'On average, your peers pay (40%) of all payables between $2,500 - $10,000 by ACH.',
    ],
    data: {
      labels: ['ACH', 'Others'],
      data: {
        data: [40, 60],
        backgroundColor: ['#c1232a', '#eee'],
      },
    },
    type: 'CircleChartText',
    partnerLogo: {
      src: '/img/integrations/RPMG-Logo.png',
      position: 'float-right',
    },
    options: { cutout: 40 },
  },
  {
    id: 6,
    direction: '',
    description: [
      'Faster payments networks now enable instant payments to be sent to approximately 80% of the DDAs in the United States.',
    ],
    data: {
      value: '80%',
      text: 'DDAs',
    },
    type: 'Text',
    partnerLogo: {
      src: '/img/integrations/FasterPayments-Logo.jpeg',
      position: 'float-left',
    },
  },
  {
    id: 7,
    direction: 'flex-row-reverse',
    iconConfig: { icon: 'dvr', color: 'text-blue' },
    description: [
      'Faster payments networks now enable instant payments to be sent to approximately 80% of the DDAs in the United States.',
    ],
    data: {
      text: 'Faster Payments',
    },
    type: 'IconText',
    partnerLogo: {
      src: '/img/integrations/FasterPayments-Logo.jpeg',
      position: 'float-right',
    },
  },
  {
    id: 8,
    direction: 'flex-row-reverse',
    iconConfig: { icon: 'info', color: 'text-black' },
    description: [
      '*Does not display all payment types. For further details, please review your Account Analysis statement.',
      '*Bank Deposit Assessment fess are fess associated with FDIC.',
    ],
    data: {
      text: 'Legal',
    },
    type: 'IconText',
    partnerLogo: {},
  },
];

export const DynamicBlock = ({ type, children, ...props }) => {
  switch (type) {
    case ComponentTextTypes.Static:
      return React.cloneElement(
        <RawHTMLBlock
          text={props.text}
          {...props}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'RPMG':
    case ComponentTextTypes.DonutSelection:
    case ComponentTextTypes.Donut:
      return React.cloneElement(
        <RPMGBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'Text':
      return React.cloneElement(
        <TextBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'IconText':
    case ComponentTextTypes.IconText:
      return React.cloneElement(
        <IconTextBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'CircleChartText':
      return React.cloneElement(
        <CircleChartTextBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'Calendar':
    case ComponentTextTypes.Calendar:
      return React.cloneElement(
        <CalendarBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
  }
};
