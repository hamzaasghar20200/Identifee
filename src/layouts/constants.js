export const getClientPortalOrganization = () => {
  return JSON.parse(sessionStorage.getItem('organizationObj'));
};

export const getClientPortalToken = () => {
  return JSON.parse(sessionStorage.getItem('tokenPayload'));
};
