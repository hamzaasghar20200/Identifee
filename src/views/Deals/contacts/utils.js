import userService from '../../../services/user.service';
import {
  ADDRESS_STATE,
  ADDRESS_STREET,
  DEAL_TYPE,
  EMAIL_LOCATION,
  EMAIL_LOCATION_1,
  EMAIL_LOCATION_2,
  PHONE_LOCATION,
  PHONE_LOCATION_2,
  PHONE_LOCATION_3,
  PHONE_LOCATION_4,
  STATUS,
} from '../../../utils/constants';
import {
  initialFilters,
  initialOrgForm,
  initialPeopleForm,
  initialDealsForm,
} from './Contacts.constants';

export const onGetOwners = (
  filter,
  dataCallback,
  existingOwners,
  extraData
) => {
  async function onGetUsers() {
    const response = await userService
      .getUsers(
        { ...filter, status: 'active' },
        {
          limit: 10,
          extraData,
        }
      )
      .catch((err) => err);

    const { data } = response || {};
    let users = [];

    const addName = (u) => ({ ...u, name: `${u.first_name} ${u.last_name}` });

    if (existingOwners && Object.keys(data).length) {
      // only interested in users who are not existing owners
      const existingOwnerIds = existingOwners.map(({ user_id }) => user_id);
      users = data.users
        .filter(({ id }) => !existingOwnerIds.includes(id))
        .map(addName);
    }

    return dataCallback(
      users.length
        ? users.filter((i) => i?.first_name && i?.last_name).map(addName)
        : !existingOwners
        ? data?.users.filter((i) => i?.first_name && i?.last_name).map(addName)
        : []
    );
  }

  onGetUsers();
};

export const onInputSearch = async (e, data, setData) => {
  if (!e) {
    return setData({
      search: '',
    });
  }

  const { value } = e.target;

  setData({
    ...data,
    search: value || '',
  });
};

export const onHandleSelect = (item, name, dispatch, setSelect) => {
  if (name === STATUS || name === DEAL_TYPE) setSelect(item.title);

  if (name === 'lead_source') setSelect(item.name);

  if (name === 'label_id') {
    return dispatch({
      type: 'set',
      input: name,
      payload: item.target.value,
    });
  }

  if (name === DEAL_TYPE) {
    return dispatch({
      type: 'set',
      input: name,
      payload: item.id,
    });
  }

  if (name === 'organization_id' && item.organization_id) {
    return dispatch({
      type: 'set',
      input: name,
      payload: item.organization_id,
    });
  }

  const payloadFields = [
    EMAIL_LOCATION,
    EMAIL_LOCATION_1,
    EMAIL_LOCATION_2,
    PHONE_LOCATION,
    PHONE_LOCATION_2,
    PHONE_LOCATION_3,
    PHONE_LOCATION_4,
    ADDRESS_STATE,
    ADDRESS_STREET,
  ];

  dispatch({
    type: 'set',
    input: name,
    payload:
      payloadFields.includes(name) ||
      name.includes(PHONE_LOCATION) ||
      name.includes(EMAIL_LOCATION)
        ? item.name
        : item.id,
  });
};

export const onInputChange = (e, dispatch) => {
  const { name, value, type, checked } = e.target;

  if (name === 'amount') {
    const preg = /^([0-9]+\.?[0-9]{0,2})$/;

    if (preg.test(value)) {
      return dispatch({
        type: 'set',
        input: name,
        payload: value,
      });
    } else if (value < 2) {
      dispatch({
        type: 'set',
        input: name,
        payload: value,
      });
    }
    return;
  }
  dispatch({
    type: 'set',
    input: name,
    payload:
      type === 'number' ? Number(value) : type === 'checkbox' ? checked : value,
  });
};

export const onInputChangeAmount = (e, dispatch, data) => {
  const { name, value, type, checked } = e.target;

  if (name === 'amount') {
    const preg = /^([0-9]+\.?[0-9]{0,2})$/;

    if (preg.test(value)) {
      return dispatch({
        ...data,
        type: 'set',
        input: name,
        amount: value,
      });
    } else if (value < 2) {
      dispatch({
        ...data,
        type: 'set',
        input: name,
        amount: value,
      });
    }
    return;
  }
  dispatch({
    ...data,
    type: 'set',
    input: name,
    amount:
      type === 'number' ? Number(value) : type === 'checkbox' ? checked : value,
  });
};

export const reducer = (state, action) => {
  switch (action.type) {
    case 'load':
      return {
        ...action.payload,
      };
    case 'add_email':
    case 'add_phone':
      return {
        ...state,
        ...action.payload,
      };
    case 'set':
      return {
        ...state,
        [action.input]: action.payload,
      };
    case 'remove':
      delete state[action.input];

      return {
        ...state,
      };
    case 'reset':
      return initialFilters;
    case 'reset-peopleForm':
      return initialPeopleForm;
    case 'reset-orgForm':
      return initialOrgForm;
    case 'reset-dealForm':
      return initialDealsForm;
    default:
      return state;
  }
};

export const tableFiltersReducer = (state, action) => {
  switch (action.type) {
    case 'set':
      return {
        ...state,
        [action.input]: action.fullPayLoad,
      };
    case 'remove':
      delete state[action.input];

      return {
        ...state,
      };
    case 'reset':
      return {};
    default:
      return state;
  }
};

export const changePaginationPage = (newPage, setPaginationPage) => {
  setPaginationPage((prev) => ({ ...prev, page: newPage }));
};

export const onHandleCloseModal = (dispatch, toggle, type) => {
  dispatch({
    type,
  });

  toggle();
};

export const removeCustomFieldsFromActivityForm = (
  formValue,
  customFieldData
) => {
  const tempContactForm = { ...formValue };

  customFieldData.forEach((customField) => {
    const keyToRemove = customField.value;
    Object.entries(tempContactForm).forEach(([key, value]) => {
      if (value === keyToRemove) {
        delete tempContactForm[key];
      }
    });
  });
  return tempContactForm;
};
