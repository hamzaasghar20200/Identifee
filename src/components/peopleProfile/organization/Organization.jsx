import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { Col } from 'reactstrap';

import OrganizationCard from '../../organizationProfile/overview/OrganizationCard';
import SimpleModal from '../../modal/SimpleModal';
import DropdownSearch from '../../DropdownSearch';
import contactService from '../../../services/contact.service';
import organizationService from '../../../services/organization.service';
import stringConstants from '../../../utils/stringConstants.json';
import routes from '../../../utils/routes.json';
import { isPermissionAllowed } from '../../../utils/Utils';

const constants = stringConstants.deals.contacts.profile;

const Organization = ({
  moduleMap,
  data,
  contactId,
  getProfileInfo,
  isPrincipalOwner,
}) => {
  const [openModal, setOpenModal] = useState(false);
  const [allOrganizations, setAllOrganizations] = useState([]);
  const [selectedOrg, setSelectedOrg] = useState({});

  const handleCloseModal = () => {
    setSelectedOrg({});
    setOpenModal(false);
  };

  const getOrganizations = ({ search, limit = 5 }) => {
    setSelectedOrg({});
    organizationService
      .getOrganizations({ search }, { limit })
      .then((res) => {
        setAllOrganizations(res.data.organizations);
      })
      .catch((err) => {
        console.log(err.message);
      });
  };

  const handleSubmit = () => {
    contactService
      .linkOrganization(contactId, selectedOrg.id)
      .then(() => {
        setOpenModal(false);
        getProfileInfo();
      })
      .catch((err) => {
        console.log(err.message);
      });
  };

  useEffect(() => {
    getOrganizations({ search: '' });
  }, []);

  return (
    <div className="card mt-3">
      <div className="card-header p-3">
        {data ? (
          <Link to={`${routes.companies}/${data.id}/organization/profile`}>
            <h4 className="card-title">{moduleMap}</h4>
          </Link>
        ) : (
          <h4 className="card-title">{moduleMap}</h4>
        )}
      </div>

      <div className="card-body toggle-org py-2 px-0">
        {data ? (
          <OrganizationCard data={data} peopleContact={true} />
        ) : (
          <div>
            <SimpleModal
              onHandleCloseModal={handleCloseModal}
              open={openModal}
              buttonLabel={`Link this ${moduleMap}`}
              buttonsDisabled={!selectedOrg.id}
              handleSubmit={handleSubmit}
            >
              <DropdownSearch
                title="Search organization"
                data={allOrganizations}
                customTitle="name"
                onChange={(event) => {
                  if (event) {
                    const { value } = event.target;
                    getOrganizations({ search: value, limit: 10 });
                  }
                }}
                onHandleSelect={(item) => {
                  setSelectedOrg(item);
                }}
              />
            </SimpleModal>
            <Col className="d-flex justify-content-center flex-column align-items-center my-4">
              <p>
                {constants.noLinkedOrganization.replace(/company/g, moduleMap)}
              </p>
              {isPermissionAllowed('contacts', 'create') &&
                isPrincipalOwner && (
                  <button
                    className="btn btn-primary btn-sm"
                    onClick={() => {
                      setOpenModal(true);
                    }}
                  >
                    <span className="material-icons-outlined mr-1">add</span>
                    {constants.linkOrganizationButton.replace(
                      /company/g,
                      moduleMap
                    )}
                  </button>
                )}
            </Col>
          </div>
        )}
      </div>
    </div>
  );
};

export default Organization;
