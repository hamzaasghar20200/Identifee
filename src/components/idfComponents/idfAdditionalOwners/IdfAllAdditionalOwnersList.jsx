import { useEffect, useState, useContext } from 'react';
import { Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
import { ASSIGNED_OWNERS } from '../../../utils/constants';

import Pagination from '../../Pagination';
import ContactOwnerList from '../../peopleProfile/owners/ContactOwnerList';
import { AlertMessageContext } from '../../../contexts/AlertMessageContext';

const IdfAllAdditionalOwnersList = ({
  children,
  openAllOwners,
  setOpenAllOwners,
  service,
  serviceId,
  mainOwner,
  count,
  refreshOwners,
  setRefresOwners,
  onGetOwners,
  owners,
  isPrincipalOwner,
}) => {
  const { setSuccessMessage, setErrorMessage } =
    useContext(AlertMessageContext);
  const [newContactOwners, setNewContactOwners] = useState([]);
  const [excludeOwner, setExcludeOwner] = useState(null);
  const [pagination, setPagination] = useState({
    page: 1,
    totalPages: Math.ceil(count / 10),
    limit: 10,
  });

  useEffect(() => {
    if (service) {
      onGetOwners();
    }
  }, [pagination, refreshOwners]);

  useEffect(() => {
    if (pagination.totalPages > 1) {
      if (pagination.page === 1) {
        setExcludeOwner(owners[4]);
        const newContactOwners = owners?.slice(0, 4);

        const newContactList = [
          { user_id: mainOwner?.id, user: mainOwner, owner: true },
          ...newContactOwners,
        ];

        return setNewContactOwners(newContactList);
      }

      if (pagination.page === pagination.totalPages) {
        const newContactList = [excludeOwner, ...owners];
        return setNewContactOwners(newContactList);
      }
    }

    return setNewContactOwners(owners);
  }, [owners, mainOwner]);

  const changePage = (newPage) => {
    if (newPage <= pagination.totalPages) {
      setPagination((prev) => ({ ...prev, page: newPage }));
    }
  };

  const onRemove = async (owner) => {
    try {
      const response = await service.removeOwner(serviceId, owner.user_id);
      if (response.data?.message) {
        return setErrorMessage(response.data?.message);
      }

      if (!refreshOwners && service) onGetOwners();

      setOpenAllOwners(false);
      setSuccessMessage(
        `${owner?.user?.first_name} ${owner?.user?.last_name} as additional owner removed`
      );
      setRefresOwners && setRefresOwners(true);
    } catch (error) {
      if (error.message) {
        return setErrorMessage(error.message);
      }
      return setErrorMessage(error);
    }
  };

  const CheckContactOwner = () => {
    return newContactOwners?.map((item) => (
      <ContactOwnerList
        key={item.user_id}
        item={item}
        handleRemove={item?.owner ? null : () => onRemove(item)}
        mainOwner={mainOwner}
        isPrincipalOwner={isPrincipalOwner}
      />
    ));
  };

  return (
    <>
      {children}

      <Modal isOpen={openAllOwners} fade={false}>
        <ModalHeader
          tag="h3"
          toggle={() => setOpenAllOwners(false)}
          className="p-3"
        >
          {ASSIGNED_OWNERS}
        </ModalHeader>
        <ModalBody className="border-top mb-0 p-3">
          <div className="list-group list-group-lg list-group-flush list-group-no-gutters">
            <CheckContactOwner />
          </div>
        </ModalBody>
        {newContactOwners?.length > 10 && (
          <ModalFooter className="px-3">
            <Pagination paginationInfo={pagination} onPageChange={changePage} />
          </ModalFooter>
        )}
      </Modal>
    </>
  );
};

export default IdfAllAdditionalOwnersList;
