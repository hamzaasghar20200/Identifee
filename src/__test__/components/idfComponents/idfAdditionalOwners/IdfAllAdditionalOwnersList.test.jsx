import { fireEvent, render, screen } from '@testing-library/react';
import { waitFor } from '@testing-library/dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import IdfAllAdditionalOwnersList from '../../../../components/idfComponents/idfAdditionalOwners/IdfAllAdditionalOwnersList';
import organizationService from '../../../../services/organization.service';
import { dataGetOneOwners } from '../../../__mocks__/services/organization.service';
import stringConstants from '../../../../utils/stringConstants.json';
import { ASSIGNED_OWNERS } from '../../../../utils/constants';

jest.mock('../../../../services/organization.service');
const constants = stringConstants.deals.organizations;

describe('Test IdfAllAdditionalOwnersList', () => {
  const history = createMemoryHistory();

  const props = {
    openAllOwners: true,
    service: organizationService,
    serviceId: '1',
    count: 10,
    mainOwner: {
      id: '12345678-abcd-efgj-a1bd-d953a4bc6cc5',
      first_name: 'First Name',
      last_name: 'Last Name',
      title: 'Title',
    },
  };

  beforeEach(async () => {
    jest.clearAllMocks();
    jest.resetModules();
    jest.restoreAllMocks();
    jest.resetAllMocks();

    await organizationService.getOwners.mockImplementation(() =>
      Promise.resolve(dataGetOneOwners)
    );

    await organizationService.removeOwner.mockImplementation(() =>
      Promise.resolve({ data: {} })
    );
  });

  it('Snapshot', async () => {
    const component = render(
      <Router history={history}>
        <IdfAllAdditionalOwnersList {...props} />
      </Router>
    );

    await screen.findByText(dataGetOneOwners.data[0].user.title);

    await waitFor(() => {
      expect(component.container).toBeVisible();
      expect(component.baseElement).toMatchSnapshot();
    });
  });

  it('Shouldnt render modal', async () => {
    render(
      <Router history={history}>
        <IdfAllAdditionalOwnersList {...props} openAllOwners={false} />
      </Router>
    );

    await waitFor(() => {
      expect(document.getElementsByClassName('modal-content').length).not.toBe(
        1
      );
    });
  });

  it('Should allow call setOpenAllOwners when try close modal', async () => {
    const mockHandler = jest.fn();

    const component = render(
      <Router history={history}>
        <IdfAllAdditionalOwnersList {...props} setOpenAllOwners={mockHandler} />
      </Router>
    );

    const close = component.getByTestId('close-modal');
    fireEvent.click(close);

    await waitFor(() => {
      expect(component.container).toBeVisible();
      expect(mockHandler).toHaveBeenCalledTimes(1);
    });
  });

  it('Should render the owners with their attributes', async () => {
    const component = render(
      <Router history={history}>
        <IdfAllAdditionalOwnersList {...props} />
      </Router>
    );

    const mainOwner = await screen.findByTestId(`item-${props.mainOwner.id}`);
    const ownerAdd = await screen.findByTestId(
      `item-${dataGetOneOwners.data[0].user_id}`
    );

    await waitFor(() => {
      expect(component.container).toBeVisible();
      expect(mainOwner).toHaveTextContent('FL');
      expect(mainOwner).toHaveTextContent(props.mainOwner.first_name);
      expect(mainOwner).toHaveTextContent(props.mainOwner.last_name);
      expect(mainOwner).toHaveTextContent(props.mainOwner.title);
      expect(ownerAdd).toHaveTextContent('OA');
      expect(ownerAdd).toHaveTextContent(
        dataGetOneOwners.data[0].user.first_name
      );
      expect(ownerAdd).toHaveTextContent(
        dataGetOneOwners.data[0].user.last_name
      );
      expect(ownerAdd).toHaveTextContent(dataGetOneOwners.data[0].user.title);
    });
  });

  it('Should allow delete the additional owner', async () => {
    const mockHandler = jest.fn();

    const component = render(
      <Router history={history}>
        <IdfAllAdditionalOwnersList {...props} setRefresOwners={mockHandler} />
      </Router>
    );

    const ownerAdd = await screen.findByTestId(
      `item-${dataGetOneOwners.data[0].user_id}`
    );

    const remove = ownerAdd.getElementsByClassName(
      'material-icons-outlined'
    )[0];
    fireEvent.click(remove);

    await organizationService.getOwners.mockImplementation(() =>
      Promise.resolve({ data: [] })
    );

    const buttonDelete = screen.getByText(constants.acceptConfirmation);
    fireEvent.click(buttonDelete);

    await screen.findByText(ASSIGNED_OWNERS);

    await waitFor(() => {
      expect(component.container).toBeVisible();
      expect(ownerAdd).not.toBeInTheDocument();
      expect(mockHandler).toHaveBeenCalledTimes(1);
    });
  });
});
