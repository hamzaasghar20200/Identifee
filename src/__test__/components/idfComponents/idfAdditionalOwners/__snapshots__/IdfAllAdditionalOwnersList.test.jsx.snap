// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test IdfAllAdditionalOwnersList Snapshot 1`] = `
<body
  class="modal-open"
>
  <div />
  <div
    style="position: relative; z-index: 1050; display: block;"
    tabindex="-1"
  >
    <div
      class=""
    >
      <div
        class="modal fade"
        role="dialog"
        style="display: block;"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h2
                class="modal-title"
              >
                Assigned Owners
              </h2>
              <button
                class="close"
                data-testid="close-modal"
              >
                Ã—
              </button>
            </div>
            <div
              class="modal-body"
            >
              <div
                class="list-group list-group-lg list-group-flush list-group-no-gutters"
              >
                <div
                  class="list-group list-group-lg list-group-flush list-group-no-gutters"
                >
                  <div
                    class="list-group-item"
                    data-testid="item-12345678-abcd-efgj-a1bd-d953a4bc6cc5"
                  >
                    <div
                      class="media"
                    >
                      <div
                        class="avatar avatar-sm avatar-circle mr-3"
                      >
                        <div
                          class="avatar avatar-sm avatar-circle avatar-soft-primary "
                        >
                          <span
                            class="avatar-initials avatar-icon-font-size"
                          >
                            FL
                          </span>
                        </div>
                      </div>
                      <div
                        class="media-body"
                      >
                        <h5
                          class="mb-0"
                        >
                          <a
                            class="text-block"
                            href="/settings/profile/users/12345678-abcd-efgj-a1bd-d953a4bc6cc5"
                          >
                            First Name
                             
                            Last Name
                          </a>
                        </h5>
                        <span
                          class="text-muted font-size-sm font-weight-medium mt-1"
                        >
                          Title
                        </span>
                        <span
                          class="d-block text-muted font-size-sm"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="list-group list-group-lg list-group-flush list-group-no-gutters"
                >
                  <div
                    class="list-group-item"
                    data-testid="item-1"
                  >
                    <div
                      class="media"
                    >
                      <div
                        class="avatar avatar-sm avatar-circle mr-3"
                      >
                        <div
                          class="avatar avatar-sm avatar-circle avatar-soft-primary "
                        >
                          <span
                            class="avatar-initials avatar-icon-font-size"
                          >
                            OA
                          </span>
                        </div>
                      </div>
                      <div
                        class="media-body"
                      >
                        <h5
                          class="mb-0"
                        >
                          <a
                            class="text-block"
                            href="/settings/profile/users/1"
                          >
                            Owner 1
                             
                            Additional
                          </a>
                        </h5>
                        <span
                          class="text-muted font-size-sm font-weight-medium mt-1"
                        >
                          Title1
                        </span>
                        <span
                          class="d-block text-muted font-size-sm"
                        >
                          ownerAdditional1@hellobuild.co
                        </span>
                      </div>
                      <button
                        class="btn btn-icon btn-sm btn-ghost-danger rounded-circle"
                      >
                        <i
                          class="material-icons-outlined"
                        >
                          remove
                        </i>
                      </button>
                    </div>
                  </div>
                </div>
                <nav
                  aria-label="Page navigation example"
                  class=""
                >
                  <ul
                    class="pagination"
                  >
                    <li
                      class="page-item active"
                    >
                      <button
                        class="page-link"
                      >
                        1
                      </button>
                    </li>
                    <li
                      class="page-item"
                    >
                      <button
                        class="page-link"
                      >
                        2
                      </button>
                    </li>
                    <li
                      class="page-item"
                    >
                      <button
                        class="page-link"
                      >
                        Next
                      </button>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="modal-backdrop fade"
      />
    </div>
  </div>
</body>
`;
